specs/agents/metadata.md

```diff
@@ -136,6 +136,37 @@ For official Elastic agents, the agent name should just be the name of the langu
 
 Services running on AWS Lambda [require specific values](tracing-instrumentation-aws-lambda.md) for some of the above mentioned fields.
 
+#### Installation method
+
+Most of the APM Agents can be installed in several ways. Agents SHOULD collect information about the used installation method and send it in the `service.agent.installation.method` field within the metadata.
+
+The intention of this field is to drive telemetry so there is a way to know which installation methods are commonly used. This field MUST produce data with very low cardinality, therefore agents SHOULD use one of the values defined below.
+
+There are some well-known installation methods which can be used by multiple agents. In those cases, agents SHOULD send the following values in `service.agent.installation.method`:
+
+- `aws-lambda-layer`: when the agent runs in AWS lambda and uses [the AWS lambda extension](https://github.com/elastic/apm-aws-lambda).
+- `k8s-attach`: when the agent is attached via [the K8s webhook](https://github.com/elastic/apm-mutating-webhook).
+- `env-attach`: when the agent is installed by setting some environment variables.
+- `azure-functions`: when the agent runs within an Azure function.
+- `fleet`: when the agent is attached via fleet.
```

Comment:
`fleet` is very interesting. What would be the way to differentiate a fleet installation from the corresponding agent-specific installation method (e.g. attach-cli in case of Java) that is used underneath for fleet installation?

Reply:
Yes, we can pass a property and look for it